src/Umbraco.Web.UI.Client/src/common/services/tree.service.js

```diff
@@ -414,7 +414,12 @@ function treeService($q, treeResource, iconHelper, notificationsService, eventsS
                     for (var c = 0; c < tn.children.length; c++) {
                         if (tn.children[c].isContainer) {
                             //recurse
-                            return getTreeRoot(tn.children[c]);
+                            var root = getTreeRoot(tn.children[c]);
+
+                            //only return if we found the root in this child, otherwise continue.
+                            if(root){
```

Comment:
Isn't the problem that it isn't finding the root? There should always be a tree root

Reply:
@Shazwazza, There is a root in the second iteration in the for-loop.
It don't make sense to have a for-loop that always returns in the first iteration?

Reply:
ok cool, there's some new quirks to tree syncing since we added the 'section container' nodes, all good if it works!